### DISCUSSION TITLE
is create3_sim the right place to open hardware/firmware/ros2 issues?

### DISCUSSION THREAD

**Discussion by gbr1** [iRobotEducation/create3_docs#143] [Questions] [answered]
Hi!
I don't know which is the right place to open issues hardware related.
Is better to write here or on create3_sim?

Issues I opened are:
- https://github.com/iRobotEducation/create3_sim/issues/184
- https://github.com/iRobotEducation/create3_sim/issues/185

Thank you in advance!

**Comment by shamlian** [comment, marked as answer]
create3_sim is appropriate for bugs related to the simulator; create3_docs should be used for issues relating to the robot in general.

**Reply by gbr1** [reply]
What do you suggest to do?
Refer these issues in 2 new issues in create3_docs?
Delete that issues and open new ones in create3_docs?
Write a post here in discussion?

Thank you!

**Reply by shamlian** [reply]
I'll transfer them now.

**Reply by shamlian** [reply]
Actually -- a clarifying question -- is the parameter requested for #184 for the sim, or is it also requested for the hardware?

**Reply by gbr1** [reply]
thank you, sorry for the inconvinience.
Create3 repos structure is quite strange by a ros point of view xD

**Reply by gbr1** [reply]
> Actually -- a clarifying question -- is the parameter requested for #184 for the sim, or is it also requested for the hardware?

they are both about real robot, simulated robot is a conseguence.

**Reply by shamlian** [reply]
I am an iRobot employee, but the following statement is my own, and doesn't necessarily represent iRobot's positions, strategies, or opinions:

Since the robot is closed source, there is no repo for the robot's code, which I agree is confusing. We decided to err on the side of fewer repositories rather than generate yet another one for robot issues, but I can see the argument the other way. I'm doing my best! :) If you have constructive suggestions as to products you think do a better job organizing their material, I'm happy to take a look.

**Reply by gbr1** [reply]
Uhm, a suggestion could be to move description package to a new repo from di simulation one.
So, you could have:
- create3, a repo with description and all issues
- create3_sim, all things related to gazebo and ignition
- create3_msgs, messages (note: it is correct to put messages in a separate repo because you can use on a dev computer without installing all other stuffs)
- create3_docs for documentation, issues related to documentation

what do you think about that?

As always thank you for your support, I really love the create 3 platform and I hope it can be better and better in the time

**Reply by shamlian** [reply]
Thanks; we will review this suggestion at our next team sprint meeting. If I am understanding you correctly, the suggestion is simply to add another repository which includes the discussion forum and issues relating to the robot, instead of reusing the _docs repo. Is that correct?

**Reply by gbr1** [reply]
yes but also put in this new repo the [description](https://github.com/iRobotEducation/create3_sim/tree/main/irobot_create_common/irobot_create_description) package that is present in create3_sim because:
- it can be difficult to find
- it can be used also on real robot
